### PR TITLE
-v2: draft-ietf-ippm-ioam-data-integrity

### PR DESCRIPTION
**Pull request by IurmanJ**
Update references:
 - draft-ietf-ippm-ioam-data  is now RFC9197
 - draft-ietf-detnet-security is now RFC9055

Update keywords (RFC2119):
 - use SHOULD instead of MUST for the replay protection subsection to
relax the condition (it is not mandatory but more an advise)

Rewrite "Methods" section:
 - rewrite both encapsulating and transit node roles to make them
crystal clear
 - merge decapsulating node and Validator as they are the same now that
DEX (or equivalent) is out of scope of this document

### CODE REVIEW

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml [hunk 1]
```diff
@@ -610,49 +610,55 @@
...
+          field of the Integrity Protection subheader. Important note: if the
+          transit node does not insert IOAM-Data-Fields (e.g., it only modifies
+          IOAM-Data-Fields in-place, or does nothing), then the transit node
+          MUST NOT generate a signature and MUST NOT update the Signature field.</t>
+
+          <t>The decapsulating node (aka the Validator) is responsible for the
+          integrity verification of the IOAM-Data-Fields collected. For obvious
+          performance reasons and because it is the Validator, the decapsulating
+          node MUST NOT generate a signature based on IOAM-Data-Fields it has
+          inserted, if any, and therefore MUST NOT update the Signature field.
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data-integrity.xml:642]: I would suggest to avoid terms like "for obvious performance reasons". What might obvious today, might be really strange in 10 years time. IMHO we can exclude the note on performance reasons and simply say:

"The decapsulating node, serving as the Validator, MUST NOT generate a signature based on ..."

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml [hunk 2]
```diff
@@ -610,49 +610,55 @@
...
+          In order to validate the IOAM-Data-Fields integrity, the Validator
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data-integrity.xml:643]: s/In order to validate/To validate/

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml [hunk 3]
```diff
@@ -610,49 +610,55 @@
...
+          recomputes the signature by iteratively following the same procedure
+          as for the encapsulating and transit nodes, in that order, using their
+          respective keys (see <xref target="SymmetricMethod" /> or 
+          <xref target="AsymmetricMethod" /> depending on the approach, i.e., symmetric
+          or asymmetric). The recomputed signature is then compared to the
+          Signature field. It is trivial in some cases (e.g., with POT Type-0 or
+          E2E Option-Types), where only the encapsulating node generates a
+          signature, as specified by the method described in the current section.
+          Details on how the mapping between encapsulating nodes and keys is
+          implemented on the Validator are outside the scope of this document.
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data-integrity.xml:653]: Do you mean "..how the mapping between the node-ids of the encapsulating nodes and the keys is ..."?

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml [hunk 4]
```diff
@@ -610,49 +610,55 @@
...
+          For other cases where transit nodes also generate a signature (e.g.,
+          with Trace Option-Types), node-ids or equivalents MUST be included in
+          IOAM-Data-Fields. Details on how the mapping between node-ids and keys
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data-integrity.xml:656]: Could we remove the earlier note in line 653 that I also left a comment on and simply use this more general statement here? IMHO this statement covers the earlier one.

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml [hunk 5]
```diff
@@ -734,7 +740,7 @@
         <section title="Replay protection">
         <t>The nonce makes a signature chain unique but does not necessarily
         prevent replay attacks. To enable replay protection, both the encap node
-        and the validator MUST synchronize on a unique nonce, e.g., based on a
+        and the validator SHOULD synchronize on a unique nonce, e.g., based on a
```
Comment (brockners) [inline drafts/draft-ietf-ippm-ioam-data-integrity.xml:743]: Why the change to SHOULD? 

We might rather say that "encap node and validator MUST use a common, unique nonce" - that avoids the statement about synchronization.

**Review by brockners — COMMENTED** [review]
Hi Justin, thanks for the updates. Hope that you can address the few nits that I found. Thanks, Frank

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml (hunk 1 above)
Comment (IurmanJ) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:642]: Solved by commit [968f166](https://github.com/inband-oam/ietf/pull/252/commits/968f1662ce30060bfec5035949832775d38c9fef) in this PR

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml (hunk 4 above)
Comment (IurmanJ) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:656]: We could, but this is actually a distinct (easier) case where only the encapsulating node generates a signature. So, in that case, you wouldn't need for node-ids, which avoids issues of having both POT and a Trace (for instance) where you're not sure that the trace data fields will represent the same path as POT (or any other (future) Option-Type)... "do all nodes insert data? are they configured to do so? etc".

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml (hunk 3 above)
Comment (IurmanJ) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:653]: Discussed below in related thread

File: drafts/draft-ietf-ippm-ioam-data-integrity.xml (hunk 4 above)
Comment (IurmanJ) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:656]: We could, but this is actually a distinct (easier) case where only the encapsulating node generates a signature. So, in that case, you wouldn't need for node-ids. But I agree it would be more generic to go the way you suggest. Thoughts?

Comment (brockners) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:656]: Personally, I'm more for the "catch all" approach with a single sentence. These two sentences that were very similar in nature made me read the text multiple times. I get the nuance, but IMHO the "catch all" approach would cut it.

Comment (IurmanJ) [reply drafts/draft-ietf-ippm-ioam-data-integrity.xml:656]: It is indeed easier to read and understand. Let's do this :-)

**Comment by brockners** [comment]
LGTM. PR reflects all the discussions.
